pkg/datarepair/checker/checker.go

```diff
@@ -120,12 +122,39 @@ func lookupResponsesToNodes(responses *pb.LookupResponses) []*pb.Node {
 	return nodes
 }
 
-// Run
-func (c *checker) Run() error {
-	return nil
+// Run the checker loop
+func (c *checker) Run(ctx context.Context, interval time.Duration) error {
+	zap.S().Info("Checker is starting up")
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	var errs []error
+
+	go func() {
```

Comment:
nit -- it might be beneficial to make this a named function, outside `Run` so you can test it independently

Reply:
Or remove the goroutine completely?